lib/plugins/aws/provider.js

```diff
@@ -356,6 +361,27 @@ class AwsProvider {
             required: ['Fn::Sub'],
             additionalProperties: false,
           },
+          awsIam: {
+            type: 'object',
+            properties: {
+              role: { $ref: '#/definitions/awsIamRole' },
+              deploymentRole: { $ref: '#/definitions/awsArn' },
+            },
+            additionalProperties: false,
+          },
+          awsIamRole: {
+            anyOf: [
+              { $ref: '#/definitions/awsArnString' },
+              {
+                type: 'object',
+                properties: {
+                  managedPolicies: { $ref: '#/definitions/awsArnList' },
+                  statements: { $ref: '#/definitions/awsIamPolicyStatements' },
+                  permissionBoundary: { $ref: '#/definitions/awsArnString' },
+                },
+              },
+            ],
+          },
```

Comment:
Definitions are intended for reusable definitions. If I understand correctly both of above address specific single case. In such scenario let's put definitions directly at their schema endpoints